static/js/pivotal.js

```diff
@@ -143,21 +173,78 @@
         'X-TrackerToken': pt_token
       },
       success: function(data) {
-        // Find all git commit messages
-        var list = data.filter(function(obj) {
-          return obj.commit_type;
-        });
-        // Find repo name from comment body text
-        list = list.map(function(item) {
-          return item.text.split('https://github.com/gengo/')[1].split('/commit/')[0];
+        var PULL_REQUEST_REGEX = /Merge pull request/;
+        var COMMIT_REPO_REGEX = /https:\/\/github.com\/gengo\/(.+)\/commit\//;
+        var DEPLOY_REPO_REGEX = new RegExp('Deployed (.+) to '+ config.environment +': ');
+
+        var activities = data.filter(function(activity) {
+          return activity.commit_type === 'github' || DEPLOY_REPO_REGEX.test(activity.text);
+        }).reverse();
+
+        var activitiesGrouped = groupBy(activities, function(activity) {
+          // With commit message
+          if (COMMIT_REPO_REGEX.test(activity.text)) {
+            return activity.text.match(COMMIT_REPO_REGEX)[1];
+          }
+          // Deployed message
+          if (DEPLOY_REPO_REGEX.test(activity.text)) {
+            return activity.text.match(DEPLOY_REPO_REGEX)[1];
+          }
         });
 
-        callback(removeDupesFromArray(list));
+        var inProgress = [];
+        var deployed = [];
+        var notDeployed = [];
+        for (key in activitiesGrouped) {
+          var activity = activitiesGrouped[key][0];
+
+          // Merged repo
+          if (PULL_REQUEST_REGEX.test(activity.text)) {
+            notDeployed.push(activity.text.match(COMMIT_REPO_REGEX)[1]);
+          }
+          // In Progress repo
+          else if (COMMIT_REPO_REGEX.test(activity.text)) {
+            inProgress.push(activity.text.match(COMMIT_REPO_REGEX)[1]);
+          }
+          // Deployed repo
+          if (DEPLOY_REPO_REGEX.test(activity.text)) {
+            deployed.push(activity.text.match(DEPLOY_REPO_REGEX)[1]);
+          }
+        }
+
+        callback({
+          'all': inProgress.concat(notDeployed, deployed),
+          'in_progress': inProgress,
+          'not_deployed': notDeployed,
+          'deployed': deployed
+        });
       }
     });
   }
 
   /**
+   * getPopoverHTML return a HTML string
+   * @param  {Array} list Array of strings
+   * @return {String}     HTML text string
+   */
+  function getPopoverHTML(dependencies) {
+    // In progress
```

Comment:
If you name those variables with longer names, you won't need those comments.
Comments can easily become stale and then they can get confusing, so whenever I can I give priority to better names over more comments.